src/core/namespace.c

```diff
@@ -1925,6 +1922,52 @@ int setup_netns(const int netns_storage_socket[static 2]) {
         return r;
 }
 
+static int netns_create(const char *path) {
+        _cleanup_close_ int old_netns = -1, netns = -1;
+        int r;
+
+        r = mkdir_parents(path, 0755);
+        if (r < 0)
+                return r;
+
+        r = touch(path);
+        if (r < 0)
+                return r;
+
+        /* Save the current netns. */
+        old_netns = open("/proc/self/ns/net", O_RDONLY | O_CLOEXEC);
+        if (old_netns < 0)
+                return -errno;
+
+        /* Create a new netns */
+        if (unshare(CLONE_NEWNET) < 0)
+                return -errno;
+
+        (void) loopback_setup();
+
+        /* Mount the new netns onto the path. */
+        if (mount("/proc/self/ns/net", path, "none", MS_BIND, NULL) < 0) {
+                r = -errno;
+                (void) setns(old_netns, CLONE_NEWNET);
+                return r;
+        }
+
+        /* Open the new netns. */
+        netns = open("/proc/self/ns/net", O_RDONLY | O_CLOEXEC);
+        if (netns < 0) {
+                r = -errno;
+                (void) umount(path);
+                (void) setns(old_netns, CLONE_NEWNET);
+                return r;
+        }
+
+        /* Restore the original netns. */
+        if (setns(old_netns, CLONE_NEWNET) < 0)
+                return -errno;
+
+        return TAKE_FD(netns);
+}
+
```

Comment:
It's kinda annoying that we'd need a new option for this though. Ideally we could just handle this internally. If it's already there when we start, keep it when we exit, if it's not there, create it and destroy it on exit. Do we really need the sticky behaviour? If users want sticky, I'd say have them create the netns mount point themselves before starting networkd.

This would require pid1 to keep track of this information but that doesn't seem impossible although I don't know enough of systemd implementation-wise to really stand behind that statement.